crates/wysiwyg/src/dom/dom_methods.rs

```diff
@@ -193,14 +193,16 @@ where
 
         // Delete the nodes marked for deletion
         let deleted_handles = if !to_delete.is_empty() {
-            self.delete_nodes(to_delete.clone())
+            self.delete_nodes(to_delete)
         } else {
             Vec::new()
         };
 
-        // If our range covered multiple text-like nodes, join together
-        // the two sides of the range.
-        if range.leaves().count() > 1 {
+        if self.children().is_empty() {
+            // Nothing to join if the deletion left the DOM empty.
+        } else if range.leaves().count() > 1 {
```

Comment:
This should be a bit easier to read
```
if range.leaves.count() > 1 && !self.children().is_empty
```

Reply:
Doesn't work because we have to opt out of both the `if` and the `else if` if that statement is true, might be better to extract this to a function though to return more explicitly.